src/treetensornetworks/projttns/abstractprojttn.jl

```diff
@@ -131,13 +131,43 @@ end
 function position(
   P::AbstractProjTTN{V}, psi::TTN{V}, pos::Union{Vector{<:V},NamedEdge{V}}
 ) where {V}
-  # shift position
   P = shift_position(P, pos)
-  # remove internal edges (out of place)
+  P = invalidate_environments(P)
+  P = make_environments(P, psi)
+  return P
+end
+
+function position!(
+  P::AbstractProjTTN{V}, psi::TTN{V}, pos::Union{Vector{<:V},NamedEdge{V}}
+) where {V}
+  P = shift_position(P, pos)
+  P = invalidate_environments(P)
+  make_environments!(P, psi, e)
+  return P
+end
```

Comment:
I think we can remove this, it doesn't seem to be adding much and we could add an in-place version later if needed.

Reply:
I ended up keeping separate in-place and out-of-place versions. We can remove the in-place ones if you prefer, but I thought they may be handy to keep around just in case there is a performance difference due to the recursive calls of `make_environment(!)` (and to investigate what actually breaks due to the issue in `Dictionaries.jl`).